requirements/customer/scheduling-algorithm.tex

```diff
@@ -63,4 +63,28 @@ \section{Scheduling Algorithm}
   By default, the scheduling algorithm will \textbf{not} pause running jobs to free up resources to run non-urgent jobs.
   However, this behavior changes once the effective priority $p$ of a non-urgent job reaches a certain threshold:
   once the threshold is reached the scheduling algorithm will begin to pause non-urgent running jobs with lower effective priority $p$ if possible and necessary to run the job with an effective priority $p$ above the threshold.
-\end{enumerate}
\ No newline at end of file
+\end{enumerate}
+\begin{figure}[H]
+  \centering
+  \includegraphics[scale=0.5]{./customer/diagrams/png/main-idea.png}
+  \caption{This diagram depicts the main flow of the scheduling algorithm}
```

Comment:
Replace with the following:
This diagram depicts the high-level flow of the scheduling algorithm.
The effective priorities of jobs are reevaluated when the set of runnable jobs changes or after a periodic time interval.
The scheduling algorithm then attempts to assign jobs to work machines.
First it tries to assign all pending urgent jobs, followed by all paused non-urgent jobs, and lastly all pending non-urgent jobs.
Inside the aforementioned three steps the scheduling algorithm processes jobs in descending order of their effective priority (high priority first).
Diagrams for how the scheduling algorithm decides if a single job can be assigned/resumed are shown in graphics \<references to the other graphics\>.